openstack/utils/choose_version.go

```diff
@@ -109,3 +113,103 @@ func ChooseVersion(client *gophercloud.ProviderClient, recognized []*Version) (*
 
 	return highest, endpoint, nil
 }
+
+type SupportedMicroversions struct {
+	MaxMajor int
+	MaxMinor int
+	MinMajor int
+	MinMinor int
+}
+
+// GetSupportedMicroversions returns the minimum and maximum microversion that is supported by the ServiceClient Endpoint.
+func GetSupportedMicroversions(client *gophercloud.ServiceClient) (SupportedMicroversions, error) {
+	type valueResp struct {
+		ID         string `json:"id"`
+		Status     string `json:"status"`
+		Version    string `json:"version"`
+		MinVersion string `json:"min_version"`
+	}
+
+	type response struct {
+		Version valueResp `json:"version"`
```

Comment:
This assumes a version endpoint, like `http://example.com/compute/v2`. It fails for me locally with Ironic since I have an unversioned one, which looks like this:

```json
{
  "name": "OpenStack Ironic API",
  "description": "Ironic is an OpenStack project which enables the provision and management of baremetal machines.",
  "default_version": {
    "id": "v1",
    "links": [
      {
        "href": "http://192.168.122.1:6385/v1/",
        "rel": "self"
      }
    ],
    "status": "CURRENT",
    "min_version": "1.1",
    "version": "1.87"
  },
  "versions": [
    {
      "id": "v1",
      "links": [
        {
          "href": "http://192.168.122.1:6385/v1/",
          "rel": "self"
        }
      ],
      "status": "CURRENT",
      "min_version": "1.1",
      "version": "1.87"
    }
  ]
}
```

Reply:
Maybe we can check the `Versions` list, and if it contains only one item, assume it's the correct one? Otherwise, fail with a clear error message?

Reply:
I have updated with something I believe does what you suggested. I even added a test with a made up Ironic endpoint that returns multiple versions to check the error case. Not sure if it makes sense :sweat_smile: